### PR TITLE
fix: build-cargo-connection-string

### PR DESCRIPTION
**Pull request by flea89**
We already have CARGO secrets in Github, let's build the connection string out of those instead of having a new one.

### CODE REVIEW

File: packages/cron/src/lib/utils.js [hunk 1]
```diff
@@ -102,13 +123,34 @@ function getPgConnString (env, mode = 'rw') {
  * @param {Record<string, string|undefined>} env
  */
 export function getCargoPgPool (env) {
-  const connection = env.CARGO_PG_CONNECTION
-  if (!connection) {
+  let connectionString = env.CARGO_PG_CONNECTION
+  // if a connection string isn't specified try to build the connection string from other variables.
```
Comment (flea89) [inline packages/cron/src/lib/utils.js:127]: I've left this, just to allow for easier setup locally? Less variables to set :P
But please disagree, and we can remove this.

File: packages/cron/src/lib/utils.js [hunk 2]
```diff
@@ -92,6 +92,27 @@ function getPgConnString (env, mode = 'rw') {
...
+ * @returns
+ */
+function buildPgConnectionString ({
+  host,
+  database,
+  user,
+  password,
+  port = 5432
+}) {
+  return `postgres//${user}:${password}@${host}:${port}/${database}`
```
Comment (olizilla) [inline packages/cron/src/lib/utils.js:113]:
```suggestion
  return `postgres://${user}:${password}@${host}:${port}/${database}`
```

File: packages/cron/src/lib/utils.js (hunk 1 above)
Comment (olizilla) [reply packages/cron/src/lib/utils.js:127]: let's use the same vars in dev and prod!

File: packages/cron/src/lib/utils.js [hunk 3]
```diff
@@ -102,13 +123,30 @@ function getPgConnString (env, mode = 'rw') {
...
+  })
+
+  const connectionString = buildPgConnectionString({
+    user: env.DAG_CARGO_USER,
+    database: env.DAG_CARGO_DATABASE,
+    password: env.DAG_CARGO_PASSWORD,
+    host: env.DAG_CARGO_HOST
+  })
+
+  console.log(`Using ${connectionString}`)
```
Comment (adamalton) [inline packages/cron/src/lib/utils.js:146]: Is there a danger of this logging the DB password somewhere? Or is `console.log` only ever hooked up to something in a dev/test environment?

File: packages/cron/test/cargo.spec.js [hunk 4]
```diff
@@ -8,7 +8,10 @@ import { getCargoPgPool, getPg } from '../src/lib/utils.js'
 const env = {
   ...process.env,
   ENV: 'dev',
-  CARGO_PG_CONNECTION: 'postgres://postgres:postgres@127.0.0.1:5432/postgres?currentSchema=cargo'
+  DAG_CARGO_HOST: '127.0.0.1',
```
[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/test/cargo.spec.js:13]: I've added this explanation into a comment in the code.

**Review by adamalton — APPROVED** [review]
All looks good now.

**Pull request merged**